### ISSUE TITLE
how to remove the confidence interval ribbon of an ROC curve?

### ISSUE THREAD

**Issue by MOKA1066** [mlr-org/mlr3viz#151]
my learners performed very similar. 
the default autoplot roc curve looks like this:
`autoplot(bmr$clone(deep = T)$filter(task_id = 'tsk_rad1'), type = 'roc')`
<img width="424" alt="image" src="https://github.com/user-attachments/assets/3d27be3c-ef0e-458f-b7e3-14933e908ba9">
I would like to remove the ribbon so it looks more clear.

Thanks!

**Comment by bblodfon** [comment]
Hi @MOKA1066 , you can pass parameters in the underlying `prerec::autoplot(...)` function that can remove the confidence bands (`show_cb = FALSE`):

``` r
library(mlr3verse)
#> Loading required package: mlr3

rr = resample(task = tsk("spam"), learner = lrn("classif.log_reg", predict_type = "prob"), resampling = rsmp("cv", folds = 5))
#> INFO  [15:57:31.459] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 1/5)
#> INFO  [15:57:31.757] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 2/5)
#> INFO  [15:57:32.131] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 3/5)
#> INFO  [15:57:32.348] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 4/5)
#> INFO  [15:57:32.562] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 5/5)
#> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
#> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
#> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
#> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
#> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

autoplot(rr, type = "roc", show_cb = FALSE)
```

![](https://i.imgur.com/nLI9ngp.png)

<sup>Created on 2024-10-31 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MOKA1066** [comment]
> Hi @MOKA1066 , you can pass parameters in the underlying `prerec::autoplot(...)` function that can remove the confidence bands (`show_cb = FALSE`):
> 
> ```r
> library(mlr3verse)
> #> Loading required package: mlr3
> 
> rr = resample(task = tsk("spam"), learner = lrn("classif.log_reg", predict_type = "prob"), resampling = rsmp("cv", folds = 5))
> #> INFO  [15:57:31.459] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 1/5)
> #> INFO  [15:57:31.757] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 2/5)
> #> INFO  [15:57:32.131] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 3/5)
> #> INFO  [15:57:32.348] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 4/5)
> #> INFO  [15:57:32.562] [mlr3] Applying learner 'classif.log_reg' on task 'spam' (iter 5/5)
> #> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
> #> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
> #> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
> #> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
> #> Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
> 
> autoplot(rr, type = "roc", show_cb = FALSE)
> ```
> 
> ![](https://camo.githubusercontent.com/2ebf9ed7b69a7eae3c7e738d0503015bb18fc9b1f02273cffd912b60a259971a/68747470733a2f2f692e696d6775722e636f6d2f6e4c49396e67702e706e67)
> 
> Created on 2024-10-31 with [reprex v2.1.1](https://reprex.tidyverse.org)

hi 
Thanks for reply!
I tried show_cb = FALSE but it does not work.

<img width="939" alt="image" src="https://github.com/user-attachments/assets/fd851e07-29ca-418e-bf2d-8e8eb0910a33">

**Comment by bblodfon** [comment]
Yes, this happens because I guess you have a `bmr` variable which is a `BenchmarkResult` object I guess, for which the `...` parameters for `precrec` are not passed - see [code](https://github.com/mlr-org/mlr3viz/blob/main/R/BenchmarkResult.R#L87). For `ResampleResult` [is it](https://github.com/mlr-org/mlr3viz/blob/main/R/ResampleResult.R#L111) though, as I showed in the example, so you need to extract the `ResampleResult` from the `BenchmarkResult` (as the latter is a combination of many resamples)

Try something like:
```
rr = bmr$resample_results$resample_result[[1]]
```

**Comment by bblodfon** [comment]
@be-marc Do you remember why we don't pass the `precre` `...` parameters [in the BenchmarkResult](https://github.com/mlr-org/mlr3viz/blob/main/R/BenchmarkResult.R#L77) but do so in the [ResampleResult](https://github.com/mlr-org/mlr3viz/blob/main/R/ResampleResult.R#L111)?

**Comment by be-marc** [comment]
No. Feel free to add.

**Comment by MOKA1066** [comment]
Thanks!


rr = bmr$resample_results$resample_result[[1]]

It works. Yet I would like to draw three ROC curves in one plot without ribbon.
So I actually draw thre “rr”s:



rr1 = bmr$resample_results$resample_result[[1]]

rr2 = bmr$resample_results$resample_result[[2]]

rr3 = bmr$resample_results$resample_result[[3]]

And then, “autoplot” the “list(rr1, rr2, rr3)” in one plot without ribbon.

Is there any way to do that?

Thanks again!

On Nov 14, 2024, at 7:02 PM, John Zobolas ***@***.******@***.***>> wrote:


rr = bmr$resample_results$resample_result[[1]]

**Comment by bblodfon** [comment]
solved in #157
